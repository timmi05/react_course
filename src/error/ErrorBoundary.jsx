import React, { Component } from 'react';
import AppContainer from '../AppContainer'

export default class ErrorBoundary extends Component {
    constructor(props) {
        super(props);
        this.state = { hasError: false, error: null, errorInfo: null };
    }

    static getDerivedStateFromError(error) {
        return { hasError: true };
    }

    render() {
        if (this.state.hasError) {
            return <h1>Something went wrong :(</h1>;
        }
        return this.props.children;
    }
}