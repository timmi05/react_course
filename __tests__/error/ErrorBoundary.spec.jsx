import React from 'react';
import { shallow } from 'enzyme'
import { ErrorBoundary } from "../../src/error/ErrorBoundary";

describe('ErrorBoundary rendering', () => {
    it('renders correctly', () => {
        const component = shallow(<ErrorBoundary/>);
        expect(component).toMatchSnapshot();
    });
});
