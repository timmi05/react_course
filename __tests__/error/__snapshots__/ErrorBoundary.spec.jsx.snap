// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary rendering renders correctly 1`] = `""`;
